fix(image): start laravel horizon to process job queue

holysoles opened the pull request:
Pixelfed requires a worker to process the job queue. Documentation outlines methods to run either Laravel Horizon or Queue workers in the background: https://docs.pixelfed.org/running-pixelfed/installation.html#job-queueing

Currently the image does not run Horizon. By default, the Horizon dashboard looks like this:

![image](https://github.com/user-attachments/assets/38d6b1b8-e9df-4544-ac64-c0f5f4f6735f)

I first noticed this was an issue when trying to delete a post, the media would delete but not the post itself. Starting Horizon manually and re-deleting the post resolved the issue.

I tested this update to the Dockerfile, and we can see Horizon starts now:

![image](https://github.com/user-attachments/assets/6aba3c57-6894-43f2-b189-4989b085c75b)

holysoles commented:
image is available here if you want to test: `ghcr.io/holysoles/docker-pixelfed:dev-build001`

mattlqx commented:
I haven't published my Helm chart, but to allow for distinct scaling of frontend from the queue worker, I run these as separate deployments where the frontend has no command override and the queue worker specifies the horizon start command. So it's not desirable to automatically start it. Though, you could gate it to an env var like `HORIZON_ENABLE` or something and that'd be fine.

holysoles commented:
@mattlqx gotcha, appreciate the background! I'm new to deploying PHP/Laravel apps and will probably look into doing it that way as well then.

Updated the script as specified and also dropped some notes in the readme that might be useful for folks

mattlqx commented on start.nginx.sh:
```diff
@@ -25,4 +25,7 @@ php artisan instance:actor
 # Finally run everything
 php-fpm -D &
-nginx -g 'daemon off;'
+nginx -g 'daemon off;' &
+if [ "${ENABLE_HORIZON:-false}" = "true" ]; then
+    php artisan horizon
+fi
```
Did you test this both enabled and not? I think if it's not enabled, there will be no process in the foreground and the container will terminate.

You probably just want to rearrange this such that the last execution will be nginx in the foreground...

```suggestion
if [ "${ENABLE_HORIZON:-false}" = "true" ]; then
    php artisan horizon &
fi
nginx -g 'daemon off;'
```

holysoles replied:
i did test as written (tailing &) inside the container and it surprisingly worked, but I like your suggestion better. addressed in my latest commit

mattlqx commented on README.md:
```diff
@@ -10,6 +10,20 @@ Available tags:
+| -------- | ----------- | ------- |
+| APP_PORT | Port that nginx will listen on | - |
+| APP_DOMAIN | Server name/domain for nginx configuration | - |
+| COMPOSER_VERSION | Version of Composer to install | 2.5.5 |
+| COMPOSER_HOME | Path to Composer home directory | /var/www/.composer |
+| COMPOSER_MEMORY_LIMIT | Memory limit for Composer | -1 |
+| DEBIAN_FRONTEND | Debian frontend configuration | noninteractive |
```
These are image build args. They're only used to setup the environment during the image build from the Dockerfile. The other env vars are used at runtime by the start script.

It's fine to include the build args as a separate table for anyone that wants to build their own image from this repo.

mattlqx replied:
Actually on second look, I'm wrong. They're `ENV`s so they get set as default variables during runtime. Only `COMPOSER_VERSION` looks like it would apply during the build and if we want that to be toggled, it would need swapped to an `ARG`.

holysoles replied:
hmm you make a good point that we probably wouldnt expect people to override them..

Should I just do `APP_PORT`, `APP_DOMAIN`, and `ENABLE_HORIZON`?

mattlqx replied:
Sure 👍🏻

holysoles replied:
should be set!

mattlqx commented:
Thank you!
